eva/expression/function_expression.py

```diff
@@ -210,7 +210,7 @@ def _apply_function_expression(self, func: Callable, batch: Batch, **kwargs):
         cache_miss = np.full(len(batch), True)
         for idx, key in cache_keys.iterrows():
             val = self._cache.store.get(key.to_numpy())
-            results[idx] = val
+            results[idx] = val[output_cols].to_numpy() if val is not None else val
```

Comment:
Why do we need `output_cols` here?

Reply:
there is a shape mismatch when the output_cols is a subset of the fields returned by the model